Api create room

buihieu028 opened the pull request:
API cho owner tạo create room

GuchiGangz committed:
Update api.php

Bổ sung route create room

GuchiGangz committed:
add/RoomImage module

Bổ sung model RoomImage cùng với migration cho tb_room_image

Bổ sung RoomImageRepository + Interface

GuchiGangz committed:
add/CreateRoomRequest

Bố sung FormRequest cho api create room

GuchiGangz committed:
Update Room.php

Bổ sung thêm trường gender_type_id

GuchiGangz committed:
add/createNewRoom module

Bố sung createNewRoom trong RoomController

DinnieJ commented on api/app/Http/Controllers/Room/RoomController.php:
```diff
@@ -96,4 +113,48 @@ public function getFavoritesByUser(Request $request)
+    public function createNewRoom(CreateRoomRequest $request)
+    {
+
+        $owner_id = auth('owner')->user()->id;
```
taị sao k dùng auth('owner')->user()->inn để lấy inn rồi lấy id :))

DinnieJ commented on api/app/Http/Controllers/Room/RoomController.php:
```diff
@@ -96,4 +113,48 @@ public function getFavoritesByUser(Request $request)
+        $room_data = $request->only('title',
+            'room_type_id', 'price',
+            'acreage', 'description',
+            'gender_type_id', 'available', 'status');
```
available tự động lúc create là 1, tương tự với status

DinnieJ commented on api/app/Http/Requests/Room/CreateRoomRequest.php:
```diff
@@ -0,0 +1,72 @@
+            'gender_type_id.integer' => trans('responses.room.gender_type_id.integer'),
+            'gender_type_id.exists' => trans('responses.room.gender_type_id.exists'),
+            'images.required' => trans('responses.room.images.required')
+
```
spacing

GuchiGangz committed:
remove/status and available field

Loại bỏ trường status + available trong FormRequest và responses

GuchiGangz committed:
Update RoomController.php

Xử lý lại cách lấy id của inn_id bằng tài khoản chủ trọ

remove đi 2 trường status + available và set nó default là 1
